### PR TITLE
Add setting BMC roles to upgrade instructions Main

### PR DESCRIPTION
**Pull request by alvarez3-hpe**
## Summary and Scope

Add instructions for verifying that BMCs of NCNs have the Management role in HSM and setting them as Management if they are not.

## Issues and Related PRs

* Resolves [CASMINST-4503](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-4503)

### Tested on:

  * loki

### Test description:

Manually tested the commands that were added on loki 

## Risks and Mitigations

None

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 1]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+
+## Stage 0.6 - Verify NCN BMC Roles and Locks
+
+Both NCNs and their BMCs require their roles to be set to 'Management' in HSM for admin locking purposes. This is done automatically for NCNs but manually for their BMCs. For BMCs, this role may not have been set during the previous installation.
+
+1. Run the following command to ensure that the roles are properly set.
+
+    ```
+    ncn-m001# cray hsm state components bulkRole update --role Management --component-ids $(cray hsm state components list --role management --type node --format json | jq -r .Components[].ID | sed 's/n[0-9]*//' | tr '\n' ',' | sed 's/.$//')
+    ncn-m001# cray hsm state components query create --format json --component-ids $(cray hsm state components list --role management --type node --format json | jq -r .Components[].ID | sed 's/n[0-9]*//' | tr '\n' ',' | sed 's/.$//') | jq '.Components[] | "\(.ID) Role=\(.Role)"'
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:231]:
```suggestion
    ncn-m001# cray hsm state components query create --format json --component-ids \
                            $(cray hsm state components list --role management --type node --format json | \
                                jq -r .Components[].ID | sed 's/n[0-9]*//' | tr '\n' ',' | sed 's/.$//') | \
                                jq '.Components[] | "\(.ID) Role=\(.Role)"'
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 2]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+    "x3000c0s8b0 Role=Management"
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:232]:
```suggestion
    ```

    Expected output looks similar to the following:

    ```
    "x3000c0s8b0 Role=Management"
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 3]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+    "x3000c0s9b0 Role=Management"
+    "x3000c0s2b0 Role=Management"
+    "x3000c0s6b0 Role=Management"
+    "x3000c0s3b0 Role=Management"
+    "x3000c0s7b0 Role=Management"
+    "x3000c0s4b0 Role=Management"
+    "x3000c0s5b0 Role=Management"
+    ```
+
+1. Run the following command to ensure they're all locked (NCNs and their BMCs).
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:242]:
```suggestion
1. Run the following command to ensure that they are all locked (NCNs and their BMCs).
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 4]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
 
+<a name="lock-ncn-bmcs"></a>
+
+## Stage 0.6 - Verify NCN BMC Roles and Locks
+
+Both NCNs and their BMCs require their roles to be set to 'Management' in HSM for admin locking purposes. This is done automatically for NCNs but manually for their BMCs. For BMCs, this role may not have been set during the previous installation.
+
+1. Run the following command to ensure that the roles are properly set.
+
+    ```
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:229]:
```suggestion
    ```bash
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 5]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+    "x3000c0s6b0 Role=Management"
+    "x3000c0s3b0 Role=Management"
+    "x3000c0s7b0 Role=Management"
+    "x3000c0s4b0 Role=Management"
+    "x3000c0s5b0 Role=Management"
+    ```
+
+1. Run the following command to ensure they're all locked (NCNs and their BMCs).
+
+    ```
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:244]:
```suggestion
    ```bash
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 6]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+    ncn-m001# /opt/cray/csm/scripts/admin_access/lock_management_nodes.py
+    ```
+
+For more information about setting NCN BMC roles see [Set BMC Management Role](../../operations/hardware_state_manager/Set_BMC_Management_Role.md)
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:248]:
```suggestion
For more information about setting NCN BMC roles, see [Set BMC Management Role](../../operations/hardware_state_manager/Set_BMC_Management_Role.md).
```

File: upgrade/1.2/Stage_0_Prerequisites.md [hunk 7]
```diff
@@ -217,6 +218,37 @@ then clone a local working tree and commit appropriate changes to `customization
...
+
+For more information about locking NCNs and their BMCs see [Lock and Unlock Nodes](../../operations/hardware_state_manager/Lock_and_Unlock_Management_Nodes.md)
```
Comment (mharding-hpe) [inline upgrade/1.2/Stage_0_Prerequisites.md:250]:
```suggestion
For more information about locking NCNs and their BMCs, see [Lock and Unlock Nodes](../../operations/hardware_state_manager/Lock_and_Unlock_Management_Nodes.md).
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Also, once this PR receives the necessary approvals, we will need the backport PR to 1.2, unless this is intended for only csm-1.3.

**Review by nrockershousen — APPROVED** [review]

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**